xdsl/traits.py

```diff
@@ -444,30 +445,111 @@ def get_canonicalization_patterns(cls) -> tuple[RewritePattern, ...]:
         raise NotImplementedError()
 
 
+class EffectKind(Enum):
```

Comment:
These seem memory-specific
```suggestion
class MemoryEffectKind(Enum):
```